src/app/learnocaml_partition_view.ml

```diff
@@ -80,7 +89,36 @@ let render_classes xs =
 
 let sum_with f = List.fold_left (fun acc x -> acc + f x) 0
 
-let exercises_tab part =
+let rec students_partition students part =
+  let open Student in
+  match part with
+    |[] -> []
+    |t::q -> let student_t = List.find (fun student -> student.token = t) students in
+             student_t :: students_partition students q
+
+let list_of_students_details students part=
+  let open Student in
+  let open Partition in
+  let bad_type_students = students_partition students part.bad_type in
+  let not_graded_students = students_partition students part.not_graded in
+  let rec create_div students_list id = match students_list with 
+    | [] -> []
+    | t::q -> let tok = Token.to_string t.token in
+              let nick = Option.value t.nickname ~default:"Student" in
+              H.a ~a:[
```

Comment:
If I'm not mistaken, `H.a` is wrong here and should be `H.div` (because there is no `href=…` attribute, for example)

Reply:
Actually as `H.p` are used to contain the elements with the info of the students (cf exercises_tab), `H.div` cannot be used : a `p` element cannot contain a `div` element. Or to change that I should modify this part of the function, but this version with `a` elements seems to work well.
```ocaml
H.p (H.txt not_graded :: not_graded_students)
  :: H.p ( H.txt bad_type :: bad_type_students)
  :: H.p [H.txt total_sum]
  :: render_classes part.partition_by_grade
```

Reply:
@Plictox OK, good point! because of the `<p>`.
But what about `H.span` then? it would be cleaner I think :)

Reply:
ok done !